openy_af4_vue_app/src/components/steps/SelectActivities.vue

```diff
@@ -113,7 +113,7 @@ export default {
         }
 
         const filteredValue = activityGroup.value.filter(item => {
-          let r = false
+          let r = true
           // Items must pass both tests, so we intentionally do not ELSE these.
           // If there are excludes, the item must NOT be excluded.
           if (this.excludeByCategory.length) {
```

Comment:
This code now has no sense.
We are setting a variable to true at the beginning.
We are not changing it to false during the road and only returning that variable.
Let's double-check again if this PR changes anything.

Reply:
So the logic behind this code is the following:

## With let r = true
By default we allow any item from `activityGroup` to be passed by setting
```
let r = true
```

If the value is present in the `excludeByCategory`, then the
```
r = !this.excludeByCategory.includes(item.value.toString())
```
will result in `false`, thus not adding the item into the resulting `filteredValue` array.

Then the value is checked again using `limitByCategory` array. 
If the value is missing in this array, then the
```
r = this.limitByCategory.includes(item.value.toString())
```
will return `false` once again, ensuring it is not in the resulting `filteredValue`.

## With let r = false
Now let's consider the case where:
```
let r = false
```
And where `excludeByCategory` & `limitByCategory` are both empty, i.e. the condition
```
if (this.excludeByCategory.length)
```
and 
```
if (this.limitByCategory.length)
```
are never passed.

The code never gets to checking
`r = !this.excludeByCategory.includes(item.value.toString())` and `r = this.limitByCategory.includes(item.value.toString())`
And it never gets a chance to change the `r` to be `true`.

Reply:
Sorry, you were right.
I've missed this line of code:
`r = !this.excludeByCategory.includes(item.value.toString())`